src/lib.rs

```diff
@@ -67,6 +85,28 @@ pub struct Mode {
     pub arp_responses: ArpResponses,
 }
 
+impl Mode {
+    pub fn to_register(self) -> [u8; 1] {
+        [self.to_u8()]
+    }
+
+    pub fn to_u8(self) -> u8 {
+        let mut register = 0;
+        register |= self.on_wake_on_lan as u8;
+        register |= self.on_ping_request as u8;
+        register |= self.connection_type as u8;
+        register |= self.arp_responses as u8;
+
+        register
+    }
+}
+
+impl From<Mode> for common::Mode {
+    fn from(value: Mode) -> Self {
+        Self::Mode(value)
+    }
+}
```

Comment:
I'm not sure I see the value here. It feels like this is more code than what we previously had?

Reply:
These values represent in a good way what the device has as a Register - single register of the mode.

While the configuration of the Mode (like WOL, Force ARP) are used for the public API and represent only what the user can configure.